src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clear = 'clear';
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const target of actions) {
+    const { type, extraData, keysToRemove } = target;
+    let current;
+
+    switch (type) {
+      case add:
+        current = Object.assign({}, currentState, extraData);
+        result.push(current);
```

Comment:
```suggestion
        result.push({ ...current });
```

You can push a copy of the object to avoid overwriting the object at each iteration and push just once at the end of loop iteration